dtfs-central-api/src/services/changeStream/setupChangeStream.ts

```diff
@@ -1,19 +1,43 @@
 import { ChangeStreamDocument } from 'mongodb';
-import { getClient } from '../../drivers/db-client';
+import { getConnection } from '../../drivers/db-client';
 import { postAuditDetails } from './changeStreamApi';
 
 /**
- * Sets up a change stream on the mongodb database and sends any changes to the audit API
+ * Sets up a change stream on the mongodb database for a specific collection and sends any changes to the audit API
+ * @param collectionName Name of the collection to set up the change stream for
  * @returns
  */
-export const setupChangeStream = async () => {
-  console.info('Setting up mongodb change stream');
-  const database = await getClient();
-  const changeStream = database.watch([], { fullDocument: 'updateLookup' });
+const setupChangeStreamForCollection = async (collectionName: string) => {
+  console.info('Setting up change stream for collection', collectionName);
+  const databaseConnection = await getConnection();
+  const changeStream = (databaseConnection.collection(collectionName)).watch(
+    [{ $match: { operationType: { $in: ['insert', 'update', 'replace'] } } }, { $project: { _id: 1, fullDocument: 1, ns: 1, documentKey: 1 } }],
```

Comment:
Do we want to track `'delete'` too?
Might need handling differently in the `$project` if the `fullDocument` no longer exists

Reply:
Yeah I've removed for now, we'll probably handle delete with updating with a TTL but I've got a separate ticket now for how we can handle delete functions
Biggest issue is if we delete normally we can't have the document which will have the details of who deleted it :(